pkg/adminapi/client.go

```diff
@@ -0,0 +1,72 @@
+package adminapi
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// HTTPClientOpts defines parameters that configure an HTTP client.
+type HTTPClientOpts struct {
+	TLSSkipVerify bool
+	TLSServerName string
+	CACertPath    string
+	CACert        string
+	Headers       []string
+}
+
+// MakeHTTPClient returns an HTTP client with the specified mTLS/headers configuration.
+// BUG: This function overwrites the default transport and client in package http!
+// This problem is being left as-is during refactoring to avoid regression of untested code.
```

Comment:
This would be a good place to add some links to relevant follow up GH issues so that code readers can track the next steps needed to resolve this.

Reply:
```suggestion
// This problem is being left as-is during refactoring to avoid regression of untested code.
// https://github.com/Kong/kubernetes-ingress-controller/issues/1233
```